pkg/usersignup/usersignup.go

```diff
@@ -30,5 +34,12 @@ func TransformUsername(username string) string {
 	if strings.HasSuffix(newUsername, "-") {
 		newUsername = newUsername + "crt"
 	}
+	if len(newUsername) > maxLength {
+		newUsername = newUsername[0:maxLength]
+		if !dnsRegexp.MatchString(newUsername) {
+			// trim starting or ending hyphen
+			newUsername = strings.Trim(newUsername, "-")
+		}
+	}
```

Comment:
maybe you already saw this, if it can help there is a validation function in apimachinery for subdomain names: https://github.com/kubernetes/apimachinery/blob/8df78a92da82be9a15f1a3beaa1d7f22acae3f3d/pkg/api/validation/generic.go#L37-L42

Reply:
Thank you for sharing this, looked into it and i think we can directly use the [`IsDNS1123Label`](https://github.com/kubernetes/apimachinery/blob/8df78a92da82be9a15f1a3beaa1d7f22acae3f3d/pkg/util/validation/validation.go#L188) instead for regex.
will still have to use custom length since the function uses maxLength as 63 - which is the maxLimit of the DNSLabel.

Reply:
@mfrancisc done in [9a9de94](https://github.com/codeready-toolchain/toolchain-common/pull/309/commits/9a9de94456dbb022f7aa75a6dec18f0689fe2171)